File: Star_Trek_uri_1973.py
# Star_Trek_uri_1973.py

n = int(input())
x = [int(sheeps) for sheeps in input().strip().split()]


visited = [False for x in range(0,n)]
initial_sheeps = sum(x)

indx = 0
stolen_sheeps = 0

while 0<=indx<n:
	curr_indx = indx
	visited[curr_indx] = True
	
	indx += 1 if x[curr_indx]%2!=0 else -1
	
	if x[curr_indx]>0:
		x[curr_indx] -= 1
		stolen_sheeps += 1

print("{} {}".format(visited.count(True), initial_sheeps-stolen_sheeps))

File: Star_Trek_uri_1973.rb
# URI-1973: Star Trek 

n = gets.to_i()
x = gets.split(" ").map(&:to_i)

visited = Array.new(n, false)
initial_sheeps = x.sum()

indx = 0
stolen_sheeps = 0

while indx>=0 and indx<n do 
    visited[indx] = true 
    curr_indx = indx
    indx += x[indx].even? ? -1 : 1
    if x[curr_indx]>0 then
        x[curr_indx] -= 1
        stolen_sheeps += 1
    end
end

puts("#{visited.count(true)} #{initial_sheeps - stolen_sheeps}")
